### ISSUE TITLE
Move Read, Write, Report to the package config and make them methods

### ISSUE THREAD

**Issue by wi101** [zio/zio-config#25]
currently in the `config` package those methods are using different data type

```scala
  def read[A](config: => Config[A]): Read[A]     = Read.read[A](config)
  def write[A](config: => Config[A]): Write[A]   = Write.write[A](config)
  def report[A](config: => Config[A]): Report[A] = Report.report[A](config)
```

Move the implementation of `Read`, `Write` and `Report` to `config` and make them methods instead of data types.
those methods should return `ZIO` 

```scala
  def read[A](config: => Config[A]): ZIO[ConfigSource, List[ReadError], (ConfigReport, A)]  
  def write[A](config: => Config[A]): Map[String, String]
  def report[A](config: => Config[A]): ConfigReport
```

**Comment by afsalthaj** [comment]
@wi101 And about the methods `int`, `string` etc, we are moving to the companion object. Aren't we?

**Comment by wi101** [comment]
I think it makes sense to move them there, let's add this in the ticket

**Comment by wi101** [comment]
Well I think it wouldn't be something tricky (like in this ticket) I think we can work on that in the opened PR
I will move them :-)

**Comment by afsalthaj** [comment]
@wi101  For sure.

**Comment by pgabara** [comment]
Hi Guys, I can take this issue too as it is related to #28

**Comment by wi101** [comment]
fixed at #47

**Comment by wi101** [comment]
Will be solved at #46
